consensus.go

```diff
@@ -0,0 +1,240 @@
+package dcrlibwallet
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"sort"
+	"time"
+
+	"decred.org/dcrwallet/v2/errors"
+	w "decred.org/dcrwallet/v2/wallet"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v3"
+)
+
+type Consensus struct {
+	mwRef *MultiWallet
+	ctx   context.Context
+}
+
+func newConsensus(mwRef *MultiWallet) (*Consensus, error) {
+	c := &Consensus{
+		mwRef: mwRef,
+	}
+
+	return c, nil
+}
+
+// GetVoteChoices handles a getvotechoices request by returning configured vote
+// preferences for each agenda of the latest supported stake version for the
+// selected wallet.
+func (c *Consensus) GetVoteChoices(ctx context.Context, hash string, walletID int) (*GetVoteChoicesResult, error) {
+	wallet := c.mwRef.WalletWithID(walletID)
+	wal := wallet.Internal()
+	if wal == nil {
+		return nil, fmt.Errorf("request requires a wallet but wallet has not loaded yet")
+	}
+
+	var ticketHash *chainhash.Hash
+	if hash != "" {
+		hash, err := chainhash.NewHashFromStr(hash)
+		if err != nil {
+			return nil, fmt.Errorf("inavlid hash: %w", err)
+		}
+		ticketHash = hash
+	}
+
+	version, agendas := getAllAgendas(wal.ChainParams())
+	resp := &GetVoteChoicesResult{
+		Version: version,
+		Choices: make([]VoteChoice, len(agendas)),
+	}
+
+	choices, _, err := wal.AgendaChoices(ctx, ticketHash)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range choices {
+		resp.Choices[i] = VoteChoice{
+			AgendaID:          choices[i].AgendaID,
+			AgendaDescription: agendas[i].Vote.Description,
```

Comment:
Yeah, decrediton shows agendas from older versions as well, but the vote choices for that would default to abstain

Reply:
More like will always be abstain unless decrediton has a way of getting the actual choices the user set then because the choices returned by `wallet.AgendaChoices` are only for the current vote version. Do we know if decrediton is able to fetch the actual vote choices for past agendas?

Reply:
no indication currently of if decrediton is able to get the past vote choices